src/main/java/org/codeforamerica/messaging/services/MessageService.java

```diff
@@ -109,4 +155,9 @@ private TemplateVariant getTemplateVariant(MessageRequest messageRequest) {
         return templateVariantOptional.get();
     }
 
+    public void scheduleMessageBatch(Long messageBatchId) throws IOException {
+        MessageBatch messageBatch = messageBatchRepository.findById(messageBatchId).orElseThrow();
+        CSVReader csvReader = new CSVReader(new InputStreamReader(new ByteArrayInputStream(messageBatch.getRecipients())));
+        csvReader.stream().forEach((r) -> this.scheduleMessage(messageBatch, r) );
```

Comment:
Would it make sense in the future from a validation/response perspective to capture the `messsage` responses and do something with them? Something to discuss as part of our validation strategy.

Reply:
As this happens asynchronously and the volume could be very large, there is not much we can do that I can think of. If the message creation fails, however, we should be recording that.